src/RegistrationController.js

```diff
@@ -61,86 +65,143 @@ function (
-            }, this));
-          } else {
-            this.appState.set('username', this.get('userName'));
-            this.appState.trigger('navigate', 'signin/register-complete');
-          }
-        }, this.model));
-
-        properties.each(function(schemaProperty) {
-          var inputOptions = RegistrationFormFactory.createInputOptions(schemaProperty);
-          var subSchemas = schemaProperty.get('subSchemas');
-          var name = schemaProperty.get('name');
-          form.addInput(inputOptions);
-          if (name === 'password' && subSchemas) {
-            form.add(SubSchema.extend({id: 'subschemas-' + name, subSchemas: subSchemas}));
-          }
-        });
+      // register parse complete event listener
+      schema.on('parseComplete', function(updatedSchema) {
+        var modelProperties = updatedSchema.properties.createModelProperties();
+        // create model
+        self.model = self.createRegistrationModel(modelProperties);
+        // create form
+        var form = new Form(self.toJSON());
         var requiredFieldsLabel = Okta.View.extend({
           template: '\
           <span class="required-fields-label">{{i18n code="registration.required.fields.label" bundle="login"}}</span>\
           '
         });
         form.add(requiredFieldsLabel);
-        this.add(form);
-        this.footer = new this.Footer(this.toJSON());
-        this.add(this.footer);
-        this.addListeners();
-      }, this);
-
-      schema.fetch().then(function(){
-        postSchemaFetch();
+        // add form
+        self.add(form);
+        // add footer
+        self.footer = new self.Footer(self.toJSON());
+        self.add(self.footer);
+        self.addListeners();
+        if (updatedSchema.error) {
+          self.showErrors(updatedSchema.error, true);
+        } else {
+          // add fields
+          updatedSchema.properties.each(function(schemaProperty) {
+            var inputOptions = RegistrationFormFactory.createInputOptions(schemaProperty);
+            var subSchemas = schemaProperty.get('subSchemas');
+            var name = schemaProperty.get('name');
+            form.addInput(inputOptions);
+            if (name === 'password' && subSchemas) {
+              form.add(SubSchema.extend({id: 'subschemas-' + name, subSchemas: subSchemas}));
+            }
+          });
+        }
+      });
+      this.state.set('schema', schema);
+    },
+    getRegistrationApiUrl: function() {
+      var clientId = this.options.settings.get('clientId');
+      return this.options.settings.get('baseUrl')+'/api/v1/registration/'+clientId;
+    },
+    doPostSubmit: function () {
+      if (this.model.get('activationToken')) {
+        // register via activation token
+        this.settings.callGlobalSuccess(Enums.REGISTRATION_COMPLETE, {
+          activationToken: this.model.get('activationToken')
+        });
+        var authClient = this.model.appState.settings.authClient;
```

Comment:
Can you look into why we need to do it this way? I.e. instead of doing it manually here, let's try to use what we do elsewhere with the transaction functions (i.e. startTransaction).

Reply:
Will look into this after I am done with the refactor to move parseComplete listener into fetchInitialData.